runtime/src/iree/hal/command_buffer.c

```diff
@@ -131,6 +132,34 @@ IREE_API_EXPORT iree_string_view_t iree_hal_command_category_format(
                                      out_temp);
 }
 
+//===----------------------------------------------------------------------===//
+// iree_hal_collective_element_t
+//===----------------------------------------------------------------------===//
+
+IREE_API_EXPORT iree_device_size_t iree_hal_collective_element_byte_count(
+    iree_hal_collective_element_type_t element_type) {
+  switch (element_type) {
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_SINT_8:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_UINT_8:
+      return 1;
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_SINT_16:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_UINT_16:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_FLOAT_16:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_BFLOAT_16:
+      return 2;
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_SINT_32:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_UINT_32:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_FLOAT_32:
+      return 4;
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_SINT_64:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_UINT_64:
+    case IREE_HAL_COLLECTIVE_ELEMENT_TYPE_FLOAT_64:
+      return 8;
+    default:
+      IREE_ASSERT(false, "unhandled element type for collective op");
+  }
```

Comment:
Looks like the build complains the function does not return a value. Just return a dummy value like 4?

Reply:
That is because IREE_ASSERT is assert for the debug build, and it is empty for the release build.

Reply:
return 0 should be safe and quiet the warning